src/anbox/cmds/session_manager.cpp

```diff
@@ -222,10 +227,13 @@ anbox::cmds::SessionManager::SessionManager()
 
     // The qemu pipe is used as a very fast communication channel between guest
     // and host for things like the GLES emulation/translation, the RIL or ADB.
+    const auto injector = di::make_injector(
+        di::bind<Renderer>().to(gl_server->renderer()),
+        di::bind<Runtime>().to(rt));
     auto qemu_pipe_connector =
         std::make_shared<network::PublishedSocketConnector>(
             utils::string_format("%s/qemu_pipe", socket_path), rt,
-            std::make_shared<qemu::PipeConnectionCreator>(gl_server->renderer(), rt));
+            injector.create<std::shared_ptr<qemu::PipeConnectionCreator>>());
```

Comment:
Hey, you may want to document that new dependency injection thing, and why qemu is affected. I don't find this clear. (I'm not used to dependency injection, mind you)

Reply:
I'm afraid I don't know well enough the code base for that. Maybe adding a comment before the `const auto injector` while moving the qemu comment down would do?